src/calculate_diff_evalue.jl

```diff
@@ -1,20 +1,18 @@
-function map_to_circuit(a, ::Int, ::Union{ChebyshevSparse, Product})
+function map_to_circuit(a, ::Real, ::Union{ChebyshevSparse, Product})
 	return a
 end
 
 function map_to_circuit(a, b::Real, ::ChebyshevTower)
 	return a * b
 end
 
-function calculate_diff_evalue(feature_map_circuit::AbstractBlock, cost::AbstractBlock, var::AbstractBlock, N::Int, theta, x, pc; mapping::FeatureMap)
+function calculate_diff_evalue(DQC::DQCType, theta, x)
 	diff_evalue = 0.0
-	for j in 1:N
-		diff_evalue_pos = real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, pc, j, π/2; mapping)))
-
-		diff_evalue_neg = real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, pc, j, -π/2; mapping)))
-				
-		diff_evalue += (diff_evalue_pos - diff_evalue_neg) * map_to_circuit(gradient(_x -> phi(_x, pc, mapping), x)[1], j, mapping)
+	for j in 1:DQC.N
+		diff_evalue_pos = expect(DQC.cost, zero_state(DQC.N) => new_circuit(DQC, x, theta, j, pi/2))
+		diff_evalue_neg = expect(DQC.cost, zero_state(DQC.N) => new_circuit(DQC, x, theta, j, -pi/2))
+		diff_evalue += (diff_evalue_pos - diff_evalue_neg) * map_to_circuit(gradient(_x -> phi(_x, DQC.afm), x)[1], j, DQC.afm)
 	end
 	diff_evalue /= 2
-	return diff_evalue
+	return real(diff_evalue)
```

Comment:
why real here? Is the imaginary part

Reply:
@ChrisRackauckas Take a look at 
`diff_evalue_pos = real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, pc, j, π/2; mapping)))` (old) vs  `diff_evalue_pos = expect(DQC.cost, zero_state(DQC.N) => new_circuit(DQC, x, theta, j, pi/2))` (new)

Its more efficient to convert the final value than separate values in my opinion

Reply:
Okay! Makes sense!